### DISCUSSION TITLE
New Entity Set

### DISCUSSION THREAD

**Discussion by Cahllagerfeld** [EddieHubCommunity/api#246] [General]
![ER_Diagram_v3](https://user-images.githubusercontent.com/43843195/147882151-b72775f1-c872-4599-8123-e39f3cdc4086.png)

There should be a new Data-Structure, with a new Entity-Set:
- The User-Entity should be the leading one, with the Discord-Username as ID
- This ID is used in all other realted entities, to be able query these collections properly
- The User-Entity contains the reference to a Github-Profile, which stands in a 1:1 relation to the User-Entity.
- With the Github-Username we are able to track Github-Stats for a Github-Username and link them to the User

NOTE: The Relations in the graph are just for making it graphical, in reality we probably dont need to populate every milestone with the userdata for example.. It might be worth a thought populating the user-entity with the corresponding milestones though 🤔 

The User and GithubProfile-Entity are the most important for now. The rest can come later in subsequent Streams on the Eddiehub-Discord-Server

**Comment by Cahllagerfeld** [comment]
It might be worth a thought of linking Milestones, Socials, Standups,.... back to the User-Entity. The Object might get very big though. I'd like to hear some opinions on that.

**Reply by eddiejaoude** [reply]
Sorry for my slow reply! 

We did have a chat about this and I didn't make notes at the time. Is this the latest?

**Reply by Cahllagerfeld** [reply]
I dont think so, our plan was to keep the projects as they are and don't migrate standups, milestones, or any other linkfree data here
